### PR TITLE
Plumed

### PR DESCRIPTION
**Pull request by aradi**
Enables meta-dynamics via Plumed integration.

### CODE REVIEW

File: doc/dftb+/manual/dftbp.tex [hunk 1]
```diff
@@ -534,6 +535,12 @@ \subsection{VelocityVerlet\{\}}
 \item[\is{Masses}] If present, over-ride the atomic masses from the Slater-Koster files. See
   p.~\pref{sec:dftbp.Masses}
 
+\item[\is{Plumed}] Whether Plumed should be invoked in order to modify the
+  forces and to drive a meta-dynamics.  The parameters of the meta-dynamics must
+  be stored in the file \verb|plumed.dat| in the current directory. The file
+  must be formatted according to Plumeds own format. (Note: This option requires
```
Comment (bhourahine) [inline doc/dftb+/manual/dftbp.tex]: must be in Plumed's own format.

File: prog/dftb+/lib_dftbplus/main.F90 [hunk 2]
```diff
@@ -5262,6 +5271,48 @@ subroutine getGradients(env, sccCalc, tExtField, tXlbomd, nonSccDeriv, EField, r
...
+    !> atomic masses array
+    real(dp), intent(in), target, contiguous :: mass(:)
+
+    !> periodic?
+    logical, intent(in) :: tPeriodic
+
+    !> lattice vectors
+    real(dp), intent(in), target, contiguous :: latVecs(:,:)
+
+    derivs = -derivs
```
Comment (bhourahine) [inline prog/dftb+/lib_dftbplus/main.F90]: dervis(:,:) = -derivs 
?

File: prog/dftb+/lib_dftbplus/parser.F90 [hunk 3]
```diff
@@ -759,6 +760,12 @@ subroutine readDriver(node, parent, geom, ctrl)
       call getChildValue(node, "OutputPrefix", buffer2, "geo_end")
       ctrl%outFile = unquote(char(buffer2))
 
+      call getChildValue(node, "Plumed", ctrl%tPlumed, default=.false., child=child)
+      if (ctrl%tPlumed .and. .not. withPlumed) then
+        call detailedError(child, "Plumed can not be used since code has been compiled without&
```
Comment (bhourahine) [inline prog/dftb+/lib_dftbplus/parser.F90]: "Metadynamics can not be used since

File: prog/dftb+/lib_extlibs/plumed.F90 [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+
+    subroutine plumed_f_gcmd(key, val) bind(C, name='plumed_f_gcmd')
+      import :: c_char, c_ptr
+      character(kind=c_char), intent(in) :: key(*)
+      type(c_ptr), value :: val
+    end subroutine plumed_f_gcmd
+  end interface
+
+
+  integer :: instances = 0
```
Comment (bhourahine) [inline prog/dftb+/lib_extlibs/plumed.F90]: Comment?

File: prog/dftb+/lib_dftbplus/initprogram.F90 [hunk 5]
```diff
@@ -2198,6 +2201,23 @@ subroutine initProgramVariables(input, env)
       call init(pMDIntegrator, pVelocityVerlet)
     end if
 
+    ! Initialising plumed
+    tPlumed = input%ctrl%tPlumed
```
Comment (bhourahine) [inline prog/dftb+/lib_dftbplus/initprogram.F90]: Should probably test here if this is an internally driven MD if (tPlumed), and not some other type of calculation

Comment (aradi) [reply prog/dftb+/lib_dftbplus/initprogram.F90]: What do you exactly mean by that?

Comment (bhourahine) [reply prog/dftb+/lib_dftbplus/initprogram.F90]: If the parser is not used to initialise the input structure(Materials studio), could get here with input%ctrl%tPlumed being true for something that is not an MD run.

File: prog/dftb+/lib_dftbplus/initprogram.F90 [hunk 6]
```diff
@@ -2198,6 +2201,26 @@ subroutine initProgramVariables(input, env)
...
+      call plumedGlobalCmdVal("setNoVirial", 0)
+      call plumedGlobalCmdVal("setTimestep", deltaT)
+      call plumedGlobalCmdVal("setMDEnergyUnits", Hartree__kJ_mol)
+      call plumedGlobalCmdVal("setMDLengthUnits", Bohr__nm)
+      call plumedGlobalCmdVal("setMDTimeUnits", au__ps)
+      call plumedGlobalCmdVal("init", 0)
+    end if
+
     ! Check for extended Born-Oppenheimer MD
     tXlbomd = allocated(input%ctrl%xlbomd)
```
Comment (bhourahine) [inline prog/dftb+/lib_dftbplus/initprogram.F90]: Is XMBOMD compatible with plumed (probably not at the moment)?

**Pull request merged**